homeassistant/components/kmtronic/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The kmtronic integration."""
+import asyncio
+
+from pykmtronic.auth import Auth
+from pykmtronic.hub import KMTronicHubAPI
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, device_registry as dr
+
+from .const import (
+    CONF_HOSTNAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DATA_HOST,
+    DATA_HUB,
+    DOMAIN,
+    MANUFACTURER,
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["switch"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the kmtronic component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up kmtronic from a config entry."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    auth = Auth(
+        session,
+        f"http://{entry.data[CONF_HOSTNAME]}",
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+    )
+    hub = KMTronicHubAPI(auth)
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][entry.entry_id] = {}
+    hass.data[DOMAIN][entry.entry_id][DATA_HUB] = hub
+    hass.data[DOMAIN][entry.entry_id][DATA_HOST] = entry.data[DATA_HOST]
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        identifiers={(DOMAIN, hub.name)},
```

Comment:
`        identifiers={(DOMAIN, hub.name)},`

It looks like this isn't stable
https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
unfortunately this device is very basic, and provides no information beyond a XML containing the relay's status :(

Reply:
We can't use it as an identifier if it won't survive the host/ip being changed as doesn't meet the design requirements for identifiers. 

Is there a way to get the Serial number or MAC address ?

Reply:
no ... these devices are really dumb and basic.

There isn't any configuration possible beyond defining user/password and Static IP/DHCP

Even the http frontend is very basic! No firmware version, no nothing...

And you don't get an interface for firmware upgrades (don't know if it is even possible!). Yet they are very cheap and reliable 😬

Reply:
https://info.kmtronic.com/manuals/user_manuals/W2CR_WEB_TWO_CHANNEL_RELAY.pdf

It looks like there is some way to get the mac, but I'm not sure its possible without scraping

<img width="694" alt="Screen Shot 2021-02-21 at 9 34 52 AM" src="https://user-images.githubusercontent.com/663432/108636235-1123ba00-7428-11eb-9048-3a2911329ff3.png">

Reply:
mine doesn't have that option :(


<img width="753" alt="Screenshot 2021-02-21 at 19 47 41" src="https://user-images.githubusercontent.com/137684/108636529-c03eb080-747d-11eb-838c-cb2125a19d6c.png">